PSReadLine/BasicEditing.cs

```diff
@@ -246,6 +246,74 @@ public static void DeleteCharOrExit(ConsoleKeyInfo? key = null, object arg = nul
             _singleton.DeleteCharImpl(1, orExit: true);
         }
 
+        /// <summary>
+        /// Upcase the current word and move to the next one.
+        /// </summary>
+        public static void UpcaseWord(ConsoleKeyInfo? key = null, object arg = null)
+        {
+            if (_singleton._current >= _singleton._buffer.Length)
+            {
+                Ding();
+                return;
+            }
+
+            int wordlen = _singleton.FindForwardWordPoint(_singleton.Options.WordDelimiters) - _singleton._current;
+            string word = _singleton._buffer.ToString(_singleton._current, wordlen);
+            word = word.ToUpper();
+
+            _singleton._buffer.Remove(_singleton._current, wordlen);
+            _singleton._buffer.Insert(_singleton._current, word);
+
+            _singleton.MoveCursor(Math.Min(_singleton._current + wordlen, _singleton._buffer.Length));
+            _singleton.Render();
+        }
+
+        /// <summary>
+        /// Downcase the current word and move to the next one.
+        /// </summary>
+        public static void DowncaseWord(ConsoleKeyInfo? key = null, object arg = null)
```

Comment:
The code in this method is mostly duplicate with `UpcaseWord`. Can you create a helper method `UpdateWordCase` and call it from both `UpcaseWord` and `DowncaseWord`, using a bool flag to indicate it's for upper case or lower case?